### DISCUSSION TITLE
Attempting to install cleanmymac (4.13.6) Cask continually results in SHA256 mismatch

### DISCUSSION THREAD

**Discussion by G4nda1f** [Homebrew/discussions#4567] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.22
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9d01e1deafca35f30b709b5f6a031faaff9642ec
Last commit: 11 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 2cdd0039cb43372f0334b316920024bca3ea4f05
Core tap last commit: 4 months ago
Core tap branch: master
Core tap JSON: 12 Jun 22:47 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 14.0.0 build 1400
Git: 2.41.0 => /usr/local/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 12.6.6-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  pycparser

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  lua@5.1

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  mpv
  unison
```

### Description of issue

Attempting to install/update to the latest cleanmymac version (4.13.6) seems to always result in a SHA 256 mismatch error.  Output:

-----------
% brew install --cask cleanmymac
==> Downloading https://dl.devmate.com/com.macpaw.CleanMyMac4/CleanMyMacX.dmg
Already downloaded: /Users/mark/Library/Caches/Homebrew/downloads/12f767d49a675919c455b9361377e73d6255920b55bb319af40194e51984d04b--CleanMyMacX.dmg
Error: SHA256 mismatch
Expected: 7a87a971fec6936d559bf6e32accada7d132712e2cefdc358e3336995f4f1deb
  Actual: 9bce785518f8b89c2158c6e5b11d90c8adb18df58d86f22c060a7aadf0054194
    File: /Users/mark/Library/Caches/Homebrew/downloads/12f767d49a675919c455b9361377e73d6255920b55bb319af40194e51984d04b--CleanMyMacX.dmg
To retry an incomplete download, remove the file above.
%
-----------

This happens on both of my macOS systems so it's not confined to just one machine.  So I wondered if there was an issue with the download definition for the current cleanmymac version since installation of previous versions has worked without issue.  Also, updating everything else I have installed via homebrew works just fine so it's just cleanmymac that seems to be experiencing this issue.

Update:  I neglected to note that I did follow the error instructions by deleting the noted file and trying the install/download again.  But even after this the error persists.

Additionally, looking at the information linked by SMillerDev, it appears that in following those instructions everything I can see seems to be correct.  The version information the homebrew site references for cleanmymac is 4.13.6 and the cleanmymac website indicates that 4.13.6 is the latest version.  So I'm unsure where the issue lies.

### Relevant casks

cleanmymac

**Comment by SMillerDev** [comment]
https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match

**Reply by G4nda1f** [reply]
I updated my post.  Essentially, yes I did delete the file and re-attempted install/download but the error persists.  And I checked the homebrew version information and that seems to match what the cleanmymac site has posted for the current version.  So everything seems to be in order so I'm unsure where the issue lies.

**Reply by SMillerDev** [reply]
In that case the last paragraph applies:

> Help us by submitting a fix. If you get stumped, open an issue explaining your steps so far and where you’re having trouble.

**Reply by G4nda1f** [reply]
OK.  That sounds good.  But now I'm immediately stuck.

The "submitting a fix" link goes to a section about updating an out of date cask.  But from everything I can see, the version numbers appear to match between homebrew's definition and the cleanmymac site.  Everything seems to be in order so I guess I'm unsure if this is something that needs to be done anyway.

The open an issue if you get stumped link seems to link back here which would seem to open another post here (unless I'm mistaken).  Or perhaps I'm supposed to post this on homebrew-cask or homebrew-core which is different???  And if so, I'm unclear what would be the most appropriate area given this issue.  Indeed, I'm unclear where the issue lies.  I suppose the issue is "reproducible" in that both my macs have the issue and everything else works fine.  But is this even an issue with the cask definition or something on my end?  So anyway, if I should post something in homebrew-cask or homebrew-core, I'll be happy to submit something but am currently unsure what the most appropriate thing to do to proceed further.

**Reply by SMillerDev** [reply]
Well, if you update the hash it's still an update to the cask. Have you tried that?

**Reply by G4nda1f** [reply]
Oh?  I can do that?  Or even... I **should** do that?  Is it using the hash of the file I receive?  What if my hash is compromised in some way?  Is it advisable for my hash to become the standard for the homebrew cask?  Shouldn't this come through some certified source?  Or perhaps it does?  Clearly, I have no idea how all this works.  Anyway, the link to submitting a fix references this command:

`brew bump-cask-pr --version <new_version> <outdated_cask>`

Is this what will update the hash?  Well, I went to use this command assuming that is what updates the hash but in preparation for that I had to first attempt a reinstall of cleanmymac (since I'd finally just uninstalled it in one of my attempts to fix it).  But this time, the install worked (after it had previously failed repeatedly on both my macs).  So I guess the issue has now been resolved.  But if running the above "bump-cask-pr" command is what I should do to update the hash I'll try that if I ever run into this issue in future.  Otherwise, if I should use a different command I'd be interested to know what to use.  The page with info on submitting a fix didn't seem to explicitly mention anything about updating the hash.

**Reply by SMillerDev** [reply, marked as answer]
> Oh? I can do that? Or even... I should do that? Is it using the hash of the file I receive? What if my hash is compromised in some way? Is it advisable for my hash to become the standard for the homebrew cask? Shouldn't this come through some certified source? Or perhaps it does?

Yeah, you should do that. CI will check if you are correct when you make a pull request though, so we don't just take your word for it.

> Is this what will update the hash?

It should, yes. If only the hash changed it won't be `<new_version>` as much as `<version>` though.

> The page with info on submitting a fix didn't seem to explicitly mention anything about updating the hash.

Since you're in a better position to judge what is missing, please consider making a pull request for it.

**Reply by G4nda1f** [reply]
> Since you're in a better position to judge what is missing, please consider making a pull request for it.

Hmmm...  Never actually done a pull request so it seems I'll create a disastrous mess, but hey, I did look into the help docs to see about it.  From what I gather, doing this is summarized by:

- Create a branch (apparently need to do a fork of this if I have no permission to make a branch)
- Input additional text/changes (something noting that "even if the version doesn't change, the 'bump-cask-pr' command can be used to update the hash of the file")
- Submit the the change to be merged into the homebrew/homebrew-cask branch
- I guess this is then reviewed (and hopefully modified as needed by those that actually know what they're doing)
- Update released to the branch

**Comment by chenrui333** [comment]
fixed by https://github.com/Homebrew/homebrew-cask/pull/149029
